### DISCUSSION TITLE
Radio Input button registration issues

### DISCUSSION THREAD

**Discussion by vincentntang** [react-hook-form/react-hook-form#5704] [Q&A]
**Describe the bug**

Hi, I'm having some problems with radio input buttons and registering them through react hook form. I made a demo reproducing the issue 

https://codesandbox.io/s/recreating-choicelist-issue-public-v2-15yz2

**To Reproduce**

1. Submit form
2. it will ask for you to fill fields
3. fill fields and radio buttons
4. Submit again
5. It will ask you to fill radio buttons again
6. select new radio inputs
7. Submit
8. It will submit as intended
9. Modify the radio input buttons, **it will not update the new inputs the user selected (bug)**

**Codesandbox link (Required)** - Version 7

https://codesandbox.io/s/recreating-choicelist-issue-public-v2-15yz2

**Expected behavior**

I expected the inputs to be updated, when the form is submitted again

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by bluebill1049** [comment]
your controlled component (radio) should contains its own state, and controller's job is only to collect value and check validation.
